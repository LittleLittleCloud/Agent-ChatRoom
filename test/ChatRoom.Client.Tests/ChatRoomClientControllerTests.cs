using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using ApprovalTests;
using ApprovalTests.Namers;
using ApprovalTests.Reporters;
using ChatRoom.Client.DTO;
using ChatRoom.SDK;
using ChatRoom.Room;
using FluentAssertions;
using Microsoft.AspNetCore.Hosting;
using Microsoft.AspNetCore.Mvc;
using Microsoft.AspNetCore.TestHost;
using Microsoft.Extensions.DependencyInjection;
using Microsoft.Extensions.Hosting;
using Moq;
using Newtonsoft.Json.Linq;
using Orleans.TestingHost;
using Xunit;
using static ChatRoom.Client.Tests.ChatRoomClientControllerTests;

namespace ChatRoom.Client.Tests;

[Collection(ClusterFixtureCollection.Name)]
public class ChatRoomClientControllerTests(ClusterFixture fixture)
{
    private readonly TestCluster _cluster = fixture.Cluster;
    private readonly ClientContext _clientContext = new ClientContext()
    {
        CurrentChannel = "test",
        CurrentRoom = nameof(ChatRoomClientControllerTests),
    };

    [Fact]
    [UseReporter(typeof(DiffReporter))]
    [UseApprovalSubdirectory("ApprovalTests")]
    public async Task TestSwagger()
    {
        var webHostBuilder = new WebHostBuilder()
                         .UseEnvironment(Environments.Development) // You can set the environment you want (development, staging, production)
                         .UseStartup<Startup>(); // Startup class of your web app project

        using (var server = new TestServer(webHostBuilder))
        {
            using (var client = server.CreateClient())
            {
                string result = await client.GetStringAsync("/swagger/v1/swagger.json");
                Approvals.Verify(result);

                var schemaFile = "chatroom_client_swagger_schema.json";
                var schemaFilePath = Path.Join("Schema", schemaFile);
                var schemaJson = File.ReadAllText(schemaFilePath);
                var schema = JObject.Parse(schemaJson).ToString();
                var resultJson = JObject.Parse(result).ToString();
                resultJson.Should().BeEquivalentTo(schema);
            }
        }
    }

    [Fact]
    public async Task ItCreateAndRemoveChannelTestAsync()
    {
        var observerMock = Mock.Of<ConsoleRoomObserver>();
        var controller = new ChatRoomClientController(_cluster.Client, _clientContext, observerMock, observerMock);

        // create nameof(ChatRoomClientControllerTests) channel
        await controller.CreateChannel(new CreateChannelRequest(nameof(ChatRoomClientControllerTests)));

        var channelResponse = await controller.GetChannels();
        var channels = (channelResponse.Result as OkObjectResult)?.Value as IEnumerable<ChannelInfo>;

        channels.Should().NotBeNull();
        channels!.Any(c => c.Name == nameof(ChatRoomClientControllerTests)).Should().BeTrue();

        // delete nameof(ChatRoomClientControllerTests) channel
        await controller.DeleteChannel(new DeleteChannelRequest(nameof(ChatRoomClientControllerTests)));

        channelResponse = await controller.GetChannels();
        channels = (channelResponse.Result as OkObjectResult)?.Value as IEnumerable<ChannelInfo>;

        channels.Should().NotBeNull();
        channels!.Any(c => c.Name == nameof(ChatRoomClientControllerTests)).Should().BeFalse();
    }

    [Fact]
    public async Task ItAddAndRemoveAgentToChannelTestAsync()
    {
        var observerMock = Mock.Of<ConsoleRoomObserver>();
        var observerRef = _cluster.Client.CreateObjectReference<IRoomObserver>(observerMock);
        var controller = new ChatRoomClientController(_cluster.Client, _clientContext, observerMock, observerMock);
        var roomGrain = _cluster.Client.GetGrain<IRoomGrain>(nameof(ChatRoomClientControllerTests));
        var testAgentName = "testAgent";
        await roomGrain.AddAgentToRoom(testAgentName, "test", true, observerRef);

        var members = await controller.GetRoomMembers();
        var membersList = (members.Result as OkObjectResult)?.Value as IEnumerable<AgentInfo>;
        membersList.Should().NotBeNull();
        membersList!.Any(m => m.Name == testAgentName).Should().BeTrue();

        // test add agent to channel
        var channelName = nameof(ChatRoomClientControllerTests);
        await controller.CreateChannel(new CreateChannelRequest(channelName));
        await controller.AddAgentToChannel(new AddAgentToChannelRequest(channelName, testAgentName));

        var channelMembers = await controller.GetChannelMembers(new GetChannelMembersRequest(channelName));
        var channelMembersList = (channelMembers.Result as OkObjectResult)?.Value as IEnumerable<AgentInfo>;
        channelMembersList.Should().NotBeNull();
        channelMembersList!.Any(m => m.Name == testAgentName).Should().BeTrue();

        // test remove agent from channel
        await controller.RemoveAgentFromChannel(new RemoveAgentFromChannelRequest(channelName, testAgentName));

        channelMembers = await controller.GetChannelMembers(new GetChannelMembersRequest(channelName));
        channelMembersList = (channelMembers.Result as OkObjectResult)?.Value as IEnumerable<AgentInfo>;
        channelMembersList.Should().NotBeNull();
        channelMembersList!.Any(m => m.Name == testAgentName).Should().BeFalse();

        // test remove agent from room
        await roomGrain.RemoveAgentFromRoom(testAgentName);

        members = await controller.GetRoomMembers();
        membersList = (members.Result as OkObjectResult)?.Value as IEnumerable<AgentInfo>;
        membersList.Should().NotBeNull();
    }

    public sealed class ClusterFixture : IDisposable
    {
        public TestCluster Cluster { get; } = new TestClusterBuilder()
            .AddSiloBuilderConfigurator<TestSiloConfigurator>()
            .Build();

        public ClusterFixture() => Cluster.Deploy();

        void IDisposable.Dispose() => Cluster.StopAllSilos();
    }


    [CollectionDefinition(Name)]
    public sealed class ClusterFixtureCollection : ICollectionFixture<ClusterFixture>
    {
        public const string Name = nameof(ClusterFixtureCollection);
    }

    public sealed class TestSiloConfigurator : ISiloConfigurator
    {
        public void Configure(ISiloBuilder hostBuilder)
        {
            hostBuilder
                .AddMemoryGrainStorage("PubSubStore")
                .ConfigureServices(services =>
                {
                    services.AddSingleton<ChannelConfiguration>();
                });
        }
    }

}
