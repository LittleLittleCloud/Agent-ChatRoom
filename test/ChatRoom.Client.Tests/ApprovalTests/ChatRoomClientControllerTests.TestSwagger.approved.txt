{
  "openapi": "3.0.1",
  "info": {
    "title": "ChatRoom.Client",
    "version": "1.0"
  },
  "paths": {
    "/api/ChatRoomClient/SendTextMessageToChannel": {
      "post": {
        "tags": [
          "ChatRoomClient"
        ],
        "requestBody": {
          "content": {
            "application/json": {
              "schema": {
                "$ref": "#/components/schemas/SendTextMessageToChannelRequest"
              }
            },
            "text/json": {
              "schema": {
                "$ref": "#/components/schemas/SendTextMessageToChannelRequest"
              }
            },
            "application/*+json": {
              "schema": {
                "$ref": "#/components/schemas/SendTextMessageToChannelRequest"
              }
            }
          }
        },
        "responses": {
          "200": {
            "description": "OK"
          }
        }
      }
    },
    "/api/ChatRoomClient/GetRoomCheckpoints": {
      "get": {
        "tags": [
          "ChatRoomClient"
        ],
        "responses": {
          "200": {
            "description": "OK",
            "content": {
              "text/plain": {
                "schema": {
                  "type": "array",
                  "items": {
                    "type": "string"
                  }
                }
              },
              "application/json": {
                "schema": {
                  "type": "array",
                  "items": {
                    "type": "string"
                  }
                }
              },
              "text/json": {
                "schema": {
                  "type": "array",
                  "items": {
                    "type": "string"
                  }
                }
              }
            }
          }
        }
      }
    },
    "/api/ChatRoomClient/UnloadCheckpoint": {
      "get": {
        "tags": [
          "ChatRoomClient"
        ],
        "responses": {
          "200": {
            "description": "OK"
          }
        }
      }
    },
    "/api/ChatRoomClient/LoadCheckpoint": {
      "get": {
        "tags": [
          "ChatRoomClient"
        ],
        "parameters": [
          {
            "name": "checkpointName",
            "in": "query",
            "schema": {
              "type": "string"
            }
          }
        ],
        "responses": {
          "200": {
            "description": "OK"
          }
        }
      }
    },
    "/api/ChatRoomClient/SaveCheckpoint": {
      "get": {
        "tags": [
          "ChatRoomClient"
        ],
        "responses": {
          "200": {
            "description": "OK"
          }
        }
      }
    },
    "/api/ChatRoomClient/DeleteCheckpoint/{checkpointPath}": {
      "get": {
        "tags": [
          "ChatRoomClient"
        ],
        "parameters": [
          {
            "name": "checkpointPath",
            "in": "path",
            "required": true,
            "schema": {
              "type": "string"
            }
          }
        ],
        "responses": {
          "200": {
            "description": "OK"
          }
        }
      }
    },
    "/api/ChatRoomClient/GetChannels": {
      "get": {
        "tags": [
          "ChatRoomClient"
        ],
        "responses": {
          "200": {
            "description": "OK",
            "content": {
              "text/plain": {
                "schema": {
                  "type": "array",
                  "items": {
                    "$ref": "#/components/schemas/ChannelInfo"
                  }
                }
              },
              "application/json": {
                "schema": {
                  "type": "array",
                  "items": {
                    "$ref": "#/components/schemas/ChannelInfo"
                  }
                }
              },
              "text/json": {
                "schema": {
                  "type": "array",
                  "items": {
                    "$ref": "#/components/schemas/ChannelInfo"
                  }
                }
              }
            }
          }
        }
      }
    },
    "/api/ChatRoomClient/GetUserInfo": {
      "get": {
        "tags": [
          "ChatRoomClient"
        ],
        "responses": {
          "200": {
            "description": "OK",
            "content": {
              "text/plain": {
                "schema": {
                  "$ref": "#/components/schemas/AgentInfo"
                }
              },
              "application/json": {
                "schema": {
                  "$ref": "#/components/schemas/AgentInfo"
                }
              },
              "text/json": {
                "schema": {
                  "$ref": "#/components/schemas/AgentInfo"
                }
              }
            }
          }
        }
      }
    },
    "/api/ChatRoomClient/ClearHistory/{channelName}": {
      "get": {
        "tags": [
          "ChatRoomClient"
        ],
        "parameters": [
          {
            "name": "channelName",
            "in": "path",
            "required": true,
            "schema": {
              "type": "string"
            }
          }
        ],
        "responses": {
          "200": {
            "description": "OK"
          }
        }
      }
    },
    "/api/ChatRoomClient/NewMessageSse/{channelName}": {
      "get": {
        "tags": [
          "ChatRoomClient"
        ],
        "parameters": [
          {
            "name": "channelName",
            "in": "path",
            "required": true,
            "schema": {
              "type": "string"
            }
          }
        ],
        "responses": {
          "200": {
            "description": "OK"
          }
        }
      }
    },
    "/api/ChatRoomClient/GenerateNextReply": {
      "post": {
        "tags": [
          "ChatRoomClient"
        ],
        "requestBody": {
          "content": {
            "application/json": {
              "schema": {
                "$ref": "#/components/schemas/GenerateNextReplyRequest"
              }
            },
            "text/json": {
              "schema": {
                "$ref": "#/components/schemas/GenerateNextReplyRequest"
              }
            },
            "application/*+json": {
              "schema": {
                "$ref": "#/components/schemas/GenerateNextReplyRequest"
              }
            }
          }
        },
        "responses": {
          "200": {
            "description": "OK",
            "content": {
              "text/plain": {
                "schema": {
                  "$ref": "#/components/schemas/GenerateNextReplyResponse"
                }
              },
              "application/json": {
                "schema": {
                  "$ref": "#/components/schemas/GenerateNextReplyResponse"
                }
              },
              "text/json": {
                "schema": {
                  "$ref": "#/components/schemas/GenerateNextReplyResponse"
                }
              }
            }
          }
        }
      }
    },
    "/api/ChatRoomClient/GetRoomMembers": {
      "get": {
        "tags": [
          "ChatRoomClient"
        ],
        "responses": {
          "200": {
            "description": "OK",
            "content": {
              "text/plain": {
                "schema": {
                  "type": "array",
                  "items": {
                    "$ref": "#/components/schemas/AgentInfo"
                  }
                }
              },
              "application/json": {
                "schema": {
                  "type": "array",
                  "items": {
                    "$ref": "#/components/schemas/AgentInfo"
                  }
                }
              },
              "text/json": {
                "schema": {
                  "type": "array",
                  "items": {
                    "$ref": "#/components/schemas/AgentInfo"
                  }
                }
              }
            }
          }
        }
      }
    },
    "/api/ChatRoomClient/GetChannelInfo/{channelName}": {
      "get": {
        "tags": [
          "ChatRoomClient"
        ],
        "parameters": [
          {
            "name": "channelName",
            "in": "path",
            "required": true,
            "schema": {
              "type": "string"
            }
          }
        ],
        "responses": {
          "200": {
            "description": "OK",
            "content": {
              "text/plain": {
                "schema": {
                  "$ref": "#/components/schemas/ChannelInfo"
                }
              },
              "application/json": {
                "schema": {
                  "$ref": "#/components/schemas/ChannelInfo"
                }
              },
              "text/json": {
                "schema": {
                  "$ref": "#/components/schemas/ChannelInfo"
                }
              }
            }
          }
        }
      }
    },
    "/api/ChatRoomClient/EditTextMessage": {
      "post": {
        "tags": [
          "ChatRoomClient"
        ],
        "requestBody": {
          "content": {
            "application/json": {
              "schema": {
                "$ref": "#/components/schemas/EditTextMessageRequest"
              }
            },
            "text/json": {
              "schema": {
                "$ref": "#/components/schemas/EditTextMessageRequest"
              }
            },
            "application/*+json": {
              "schema": {
                "$ref": "#/components/schemas/EditTextMessageRequest"
              }
            }
          }
        },
        "responses": {
          "200": {
            "description": "OK"
          }
        }
      }
    },
    "/api/ChatRoomClient/DeleteMessage/{channelName}/{messageId}": {
      "get": {
        "tags": [
          "ChatRoomClient"
        ],
        "parameters": [
          {
            "name": "channelName",
            "in": "path",
            "required": true,
            "schema": {
              "type": "string"
            }
          },
          {
            "name": "messageId",
            "in": "path",
            "required": true,
            "schema": {
              "type": "integer",
              "format": "int64"
            }
          }
        ],
        "responses": {
          "200": {
            "description": "OK"
          }
        }
      }
    },
    "/api/ChatRoomClient/GetChannelMembers": {
      "post": {
        "tags": [
          "ChatRoomClient"
        ],
        "requestBody": {
          "content": {
            "application/json": {
              "schema": {
                "$ref": "#/components/schemas/GetChannelMembersRequest"
              }
            },
            "text/json": {
              "schema": {
                "$ref": "#/components/schemas/GetChannelMembersRequest"
              }
            },
            "application/*+json": {
              "schema": {
                "$ref": "#/components/schemas/GetChannelMembersRequest"
              }
            }
          }
        },
        "responses": {
          "200": {
            "description": "OK",
            "content": {
              "text/plain": {
                "schema": {
                  "type": "array",
                  "items": {
                    "$ref": "#/components/schemas/AgentInfo"
                  }
                }
              },
              "application/json": {
                "schema": {
                  "type": "array",
                  "items": {
                    "$ref": "#/components/schemas/AgentInfo"
                  }
                }
              },
              "text/json": {
                "schema": {
                  "type": "array",
                  "items": {
                    "$ref": "#/components/schemas/AgentInfo"
                  }
                }
              }
            }
          }
        }
      }
    },
    "/api/ChatRoomClient/GetChannelChatHistory": {
      "post": {
        "tags": [
          "ChatRoomClient"
        ],
        "requestBody": {
          "content": {
            "application/json": {
              "schema": {
                "$ref": "#/components/schemas/GetChannelChatHistoryRequest"
              }
            },
            "text/json": {
              "schema": {
                "$ref": "#/components/schemas/GetChannelChatHistoryRequest"
              }
            },
            "application/*+json": {
              "schema": {
                "$ref": "#/components/schemas/GetChannelChatHistoryRequest"
              }
            }
          }
        },
        "responses": {
          "200": {
            "description": "OK",
            "content": {
              "text/plain": {
                "schema": {
                  "type": "array",
                  "items": {
                    "$ref": "#/components/schemas/ChatMsg"
                  }
                }
              },
              "application/json": {
                "schema": {
                  "type": "array",
                  "items": {
                    "$ref": "#/components/schemas/ChatMsg"
                  }
                }
              },
              "text/json": {
                "schema": {
                  "type": "array",
                  "items": {
                    "$ref": "#/components/schemas/ChatMsg"
                  }
                }
              }
            }
          }
        }
      }
    },
    "/api/ChatRoomClient/CreateChannel": {
      "post": {
        "tags": [
          "ChatRoomClient"
        ],
        "requestBody": {
          "content": {
            "application/json": {
              "schema": {
                "$ref": "#/components/schemas/CreateChannelRequest"
              }
            },
            "text/json": {
              "schema": {
                "$ref": "#/components/schemas/CreateChannelRequest"
              }
            },
            "application/*+json": {
              "schema": {
                "$ref": "#/components/schemas/CreateChannelRequest"
              }
            }
          }
        },
        "responses": {
          "200": {
            "description": "OK"
          }
        }
      }
    },
    "/api/ChatRoomClient/JoinChannel": {
      "post": {
        "tags": [
          "ChatRoomClient"
        ],
        "requestBody": {
          "content": {
            "application/json": {
              "schema": {
                "$ref": "#/components/schemas/JoinChannelRequest"
              }
            },
            "text/json": {
              "schema": {
                "$ref": "#/components/schemas/JoinChannelRequest"
              }
            },
            "application/*+json": {
              "schema": {
                "$ref": "#/components/schemas/JoinChannelRequest"
              }
            }
          }
        },
        "responses": {
          "200": {
            "description": "OK"
          }
        }
      }
    },
    "/api/ChatRoomClient/LeaveChannel": {
      "post": {
        "tags": [
          "ChatRoomClient"
        ],
        "requestBody": {
          "content": {
            "application/json": {
              "schema": {
                "$ref": "#/components/schemas/LeaveChannelRequest"
              }
            },
            "text/json": {
              "schema": {
                "$ref": "#/components/schemas/LeaveChannelRequest"
              }
            },
            "application/*+json": {
              "schema": {
                "$ref": "#/components/schemas/LeaveChannelRequest"
              }
            }
          }
        },
        "responses": {
          "200": {
            "description": "OK"
          }
        }
      }
    },
    "/api/ChatRoomClient/DeleteChannel": {
      "post": {
        "tags": [
          "ChatRoomClient"
        ],
        "requestBody": {
          "content": {
            "application/json": {
              "schema": {
                "$ref": "#/components/schemas/DeleteChannelRequest"
              }
            },
            "text/json": {
              "schema": {
                "$ref": "#/components/schemas/DeleteChannelRequest"
              }
            },
            "application/*+json": {
              "schema": {
                "$ref": "#/components/schemas/DeleteChannelRequest"
              }
            }
          }
        },
        "responses": {
          "200": {
            "description": "OK"
          }
        }
      }
    },
    "/api/ChatRoomClient/AddAgentToChannel": {
      "post": {
        "tags": [
          "ChatRoomClient"
        ],
        "requestBody": {
          "content": {
            "application/json": {
              "schema": {
                "$ref": "#/components/schemas/AddAgentToChannelRequest"
              }
            },
            "text/json": {
              "schema": {
                "$ref": "#/components/schemas/AddAgentToChannelRequest"
              }
            },
            "application/*+json": {
              "schema": {
                "$ref": "#/components/schemas/AddAgentToChannelRequest"
              }
            }
          }
        },
        "responses": {
          "200": {
            "description": "OK"
          }
        }
      }
    },
    "/api/ChatRoomClient/RemoveAgentFromChannel": {
      "post": {
        "tags": [
          "ChatRoomClient"
        ],
        "requestBody": {
          "content": {
            "application/json": {
              "schema": {
                "$ref": "#/components/schemas/RemoveAgentFromChannelRequest"
              }
            },
            "text/json": {
              "schema": {
                "$ref": "#/components/schemas/RemoveAgentFromChannelRequest"
              }
            },
            "application/*+json": {
              "schema": {
                "$ref": "#/components/schemas/RemoveAgentFromChannelRequest"
              }
            }
          }
        },
        "responses": {
          "200": {
            "description": "OK"
          }
        }
      }
    },
    "/api/ChatRoomClient/GetOrchestrators": {
      "get": {
        "tags": [
          "ChatRoomClient"
        ],
        "responses": {
          "200": {
            "description": "OK",
            "content": {
              "text/plain": {
                "schema": {
                  "type": "array",
                  "items": {
                    "type": "string"
                  }
                }
              },
              "application/json": {
                "schema": {
                  "type": "array",
                  "items": {
                    "type": "string"
                  }
                }
              },
              "text/json": {
                "schema": {
                  "type": "array",
                  "items": {
                    "type": "string"
                  }
                }
              }
            }
          }
        }
      }
    },
    "/api/ChatRoomClient/GetChannelOrchestrators": {
      "get": {
        "tags": [
          "ChatRoomClient"
        ],
        "parameters": [
          {
            "name": "channel",
            "in": "query",
            "schema": {
              "type": "string"
            }
          }
        ],
        "responses": {
          "200": {
            "description": "OK",
            "content": {
              "text/plain": {
                "schema": {
                  "type": "array",
                  "items": {
                    "type": "string"
                  }
                }
              },
              "application/json": {
                "schema": {
                  "type": "array",
                  "items": {
                    "type": "string"
                  }
                }
              },
              "text/json": {
                "schema": {
                  "type": "array",
                  "items": {
                    "type": "string"
                  }
                }
              }
            }
          }
        }
      }
    },
    "/api/ChatRoomClient/AddOrchestratorToChannel": {
      "post": {
        "tags": [
          "ChatRoomClient"
        ],
        "requestBody": {
          "content": {
            "application/json": {
              "schema": {
                "$ref": "#/components/schemas/AddOrchestratorToChannelRequest"
              }
            },
            "text/json": {
              "schema": {
                "$ref": "#/components/schemas/AddOrchestratorToChannelRequest"
              }
            },
            "application/*+json": {
              "schema": {
                "$ref": "#/components/schemas/AddOrchestratorToChannelRequest"
              }
            }
          }
        },
        "responses": {
          "200": {
            "description": "OK"
          }
        }
      }
    },
    "/api/ChatRoomClient/RemoveOrchestratorFromChannel": {
      "post": {
        "tags": [
          "ChatRoomClient"
        ],
        "requestBody": {
          "content": {
            "application/json": {
              "schema": {
                "$ref": "#/components/schemas/RemoveOrchestratorFromChannelRequest"
              }
            },
            "text/json": {
              "schema": {
                "$ref": "#/components/schemas/RemoveOrchestratorFromChannelRequest"
              }
            },
            "application/*+json": {
              "schema": {
                "$ref": "#/components/schemas/RemoveOrchestratorFromChannelRequest"
              }
            }
          }
        },
        "responses": {
          "200": {
            "description": "OK"
          }
        }
      }
    },
    "/api/ChatRoomClient/CloneChannel": {
      "post": {
        "tags": [
          "ChatRoomClient"
        ],
        "requestBody": {
          "content": {
            "application/json": {
              "schema": {
                "$ref": "#/components/schemas/CloneChannelRequest"
              }
            },
            "text/json": {
              "schema": {
                "$ref": "#/components/schemas/CloneChannelRequest"
              }
            },
            "application/*+json": {
              "schema": {
                "$ref": "#/components/schemas/CloneChannelRequest"
              }
            }
          }
        },
        "responses": {
          "200": {
            "description": "OK"
          }
        }
      }
    },
    "/api/ChatRoomClient/EditChannelName": {
      "post": {
        "tags": [
          "ChatRoomClient"
        ],
        "requestBody": {
          "content": {
            "application/json": {
              "schema": {
                "$ref": "#/components/schemas/EditChannelNameRequest"
              }
            },
            "text/json": {
              "schema": {
                "$ref": "#/components/schemas/EditChannelNameRequest"
              }
            },
            "application/*+json": {
              "schema": {
                "$ref": "#/components/schemas/EditChannelNameRequest"
              }
            }
          }
        },
        "responses": {
          "200": {
            "description": "OK"
          }
        }
      }
    }
  },
  "components": {
    "schemas": {
      "AddAgentToChannelRequest": {
        "type": "object",
        "properties": {
          "channelName": {
            "type": "string",
            "nullable": true
          },
          "agentName": {
            "type": "string",
            "nullable": true
          }
        },
        "additionalProperties": false
      },
      "AddOrchestratorToChannelRequest": {
        "type": "object",
        "properties": {
          "channelName": {
            "type": "string",
            "nullable": true
          },
          "orchestratorName": {
            "type": "string",
            "nullable": true
          }
        },
        "additionalProperties": false
      },
      "AgentInfo": {
        "type": "object",
        "properties": {
          "name": {
            "type": "string",
            "nullable": true
          },
          "selfDescription": {
            "type": "string",
            "nullable": true
          },
          "isHuman": {
            "type": "boolean"
          }
        },
        "additionalProperties": false
      },
      "ChannelInfo": {
        "type": "object",
        "properties": {
          "name": {
            "type": "string",
            "nullable": true
          },
          "members": {
            "type": "array",
            "items": {
              "$ref": "#/components/schemas/AgentInfo"
            },
            "nullable": true
          },
          "orchestrators": {
            "type": "array",
            "items": {
              "type": "string"
            },
            "nullable": true
          }
        },
        "additionalProperties": false
      },
      "ChatMsg": {
        "type": "object",
        "properties": {
          "from": {
            "type": "string",
            "nullable": true
          },
          "created": {
            "type": "string",
            "format": "date-time"
          },
          "parts": {
            "type": "array",
            "items": {
              "$ref": "#/components/schemas/ChatMsgPart"
            },
            "nullable": true
          },
          "id": {
            "type": "integer",
            "format": "int64"
          }
        },
        "additionalProperties": false
      },
      "ChatMsgPart": {
        "type": "object",
        "properties": {
          "textPart": {
            "type": "string",
            "nullable": true
          },
          "imageData": {
            "type": "string",
            "format": "byte",
            "nullable": true
          },
          "imageUrl": {
            "type": "string",
            "format": "uri",
            "nullable": true
          },
          "mimeType": {
            "type": "string",
            "nullable": true
          }
        },
        "additionalProperties": false
      },
      "CloneChannelRequest": {
        "type": "object",
        "properties": {
          "channelName": {
            "type": "string",
            "nullable": true
          },
          "newChannelName": {
            "type": "string",
            "nullable": true
          }
        },
        "additionalProperties": false
      },
      "CreateChannelRequest": {
        "type": "object",
        "properties": {
          "channelName": {
            "type": "string",
            "nullable": true
          }
        },
        "additionalProperties": false
      },
      "DeleteChannelRequest": {
        "type": "object",
        "properties": {
          "channelName": {
            "type": "string",
            "nullable": true
          }
        },
        "additionalProperties": false
      },
      "EditChannelNameRequest": {
        "type": "object",
        "properties": {
          "oldChannelName": {
            "type": "string",
            "nullable": true
          },
          "newChannelName": {
            "type": "string",
            "nullable": true
          }
        },
        "additionalProperties": false
      },
      "EditTextMessageRequest": {
        "type": "object",
        "properties": {
          "channelName": {
            "type": "string",
            "nullable": true
          },
          "messageId": {
            "type": "integer",
            "format": "int64"
          },
          "newText": {
            "type": "string",
            "nullable": true
          }
        },
        "additionalProperties": false
      },
      "GenerateNextReplyRequest": {
        "type": "object",
        "properties": {
          "channelName": {
            "type": "string",
            "nullable": true
          },
          "chatMsgs": {
            "type": "array",
            "items": {
              "$ref": "#/components/schemas/ChatMsg"
            },
            "nullable": true
          },
          "candidates": {
            "type": "array",
            "items": {
              "type": "string"
            },
            "nullable": true
          },
          "orchestrator": {
            "type": "string",
            "nullable": true
          }
        },
        "additionalProperties": false
      },
      "GenerateNextReplyResponse": {
        "type": "object",
        "properties": {
          "message": {
            "$ref": "#/components/schemas/ChatMsg"
          }
        },
        "additionalProperties": false
      },
      "GetChannelChatHistoryRequest": {
        "type": "object",
        "properties": {
          "channelName": {
            "type": "string",
            "nullable": true
          },
          "count": {
            "type": "integer",
            "format": "int32"
          }
        },
        "additionalProperties": false
      },
      "GetChannelMembersRequest": {
        "type": "object",
        "properties": {
          "channelName": {
            "type": "string",
            "nullable": true
          }
        },
        "additionalProperties": false
      },
      "JoinChannelRequest": {
        "type": "object",
        "properties": {
          "channelName": {
            "type": "string",
            "nullable": true
          },
          "createIfNotExists": {
            "type": "boolean"
          }
        },
        "additionalProperties": false
      },
      "LeaveChannelRequest": {
        "type": "object",
        "properties": {
          "channelName": {
            "type": "string",
            "nullable": true
          }
        },
        "additionalProperties": false
      },
      "RemoveAgentFromChannelRequest": {
        "type": "object",
        "properties": {
          "channelName": {
            "type": "string",
            "nullable": true
          },
          "agentName": {
            "type": "string",
            "nullable": true
          }
        },
        "additionalProperties": false
      },
      "RemoveOrchestratorFromChannelRequest": {
        "type": "object",
        "properties": {
          "channelName": {
            "type": "string",
            "nullable": true
          },
          "orchestratorName": {
            "type": "string",
            "nullable": true
          }
        },
        "additionalProperties": false
      },
      "SendTextMessageToChannelRequest": {
        "type": "object",
        "properties": {
          "channelName": {
            "type": "string",
            "nullable": true
          },
          "message": {
            "$ref": "#/components/schemas/ChatMsg"
          }
        },
        "additionalProperties": false
      }
    }
  }
}